src/Compilers/Core/Portable/MetadataReader/PEModule.cs

```diff
@@ -3638,6 +3639,32 @@ private ConstantValue GetConstantValueOrThrow(ConstantHandle handle)
 
             matchedName = null;
             return (FirstIndex: -1, SecondIndex: -1);
+
+            void ensureCaseInsensitiveDictionary()
+            {
+                if (_lazyCaseInsensitiveForwardedTypesToAssemblyIndexMap != null)
+                {
+                    return;
+                }
+
+                var caseInsensitiveMap = new Dictionary<string, OneOrMany<(int FirstIndex, int SecondIndex)>>(StringComparer.OrdinalIgnoreCase);
```

Comment:
>OneOrMany [](http://example.com/codeflow?start=64&length=9)

Instead of keeping a OneOrMany, couldn't we just keep the first one? (the caller only uses values[0])